addons/mail/models/mail_thread.py

```diff
@@ -2885,8 +2894,21 @@ def _message_create(self, values_list):
             for x in ('from', 'to', 'cc'):
                 create_values.pop(x, None)
             create_values['partner_ids'] = [Command.link(pid) for pid in (create_values.get('partner_ids') or [])]
+            channel_ids = create_values.pop('channel_ids', None)
             create_values_list.append(create_values)
 
+            # Create message in mentioned channels
+            if channel_ids:
+                user = f'<a href="/odoo/res.partner/{self.env.user.partner_id.id}">{self.env.user.name}</a>'
+                backlink_html = f'<a href="/odoo/{self._name}/{self.id}">{create_values.get("record_name")}</a>'
+                create_channel_values = dict(create_values)
+                create_channel_values['body'] = f"{user} mentioned this channel in {backlink_html}"
+                create_channel_values['model'] = 'discuss.channel'
+                create_channel_list = [{**create_channel_values, 'res_id': channel_id} for channel_id in channel_ids]
+                messages_in_channel = self.env['mail.message'].with_context(
+                    clean_context(self.env.context)
+                ).create(create_channel_list)
+                self._notify_thread_by_channel(messages_in_channel)
```

Comment:
IMO using the `message_post` method of `discuss.channel` can create a cyclic loop because `discuss.channel` inherits from `mail.thread`. When `discuss.channel` calls `message_post`, it may end up calling the same method in `mail.thread` again.

Reply:
There is only a loop if `channel_ids` is given to the sub-call, which should never be the case (it's programmatically called here, we know what we send).